plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/service/QueryGroupPersistenceService.java

```diff
@@ -0,0 +1,165 @@
+        this.clusterService = clusterService;
+        this.deleteQueryGroupThrottlingKey = clusterService.registerClusterManagerTask(DELETE_QUERY_GROUP_THROTTLING_KEY, true);
+        setMaxQueryGroupCount(MAX_QUERY_GROUP_COUNT.get(settings));
+        clusterSettings.addSettingsUpdateConsumer(MAX_QUERY_GROUP_COUNT, this::setMaxQueryGroupCount);
+    }
+
+    /**
+     * Set maxQueryGroupCount to be newMaxQueryGroupCount
+     * @param newMaxQueryGroupCount - the max number of QueryGroup allowed
+     */
+    public void setMaxQueryGroupCount(int newMaxQueryGroupCount) {
+        validateMaxQueryGroupCount(newMaxQueryGroupCount);
+        this.maxQueryGroupCount = newMaxQueryGroupCount;
+    }
+
+    /**
+     * Validator for maxQueryGroupCount
+     * @param maxQueryGroupCount - the maxQueryGroupCount number to be verified
+     */
+    private static void validateMaxQueryGroupCount(int maxQueryGroupCount) {
+        if (maxQueryGroupCount > DEFAULT_MAX_QUERY_GROUP_COUNT_VALUE || maxQueryGroupCount < MIN_QUERY_GROUP_COUNT_VALUE) {
+            throw new IllegalArgumentException(QUERY_GROUP_COUNT_SETTING_NAME + " should be in range [1-100].");
+        }
+    }
+
+    /**
+     * Modify cluster state to delete the QueryGroup
+     * @param name - the name for QueryGroup to be deleted
+     * @param listener - ActionListener for DeleteQueryGroupResponse
+     */
+    public void deleteInClusterStateMetadata(String name, ActionListener<DeleteQueryGroupResponse> listener) {
+        clusterService.submitStateUpdateTask(SOURCE, new ClusterStateUpdateTask(Priority.NORMAL) {
+            @Override
+            public ClusterState execute(ClusterState currentState) throws Exception {
+                return deleteQueryGroupInClusterState(name, currentState);
+            }
+
+            @Override
+            public ThrottlingKey getClusterManagerThrottlingKey() {
+                return deleteQueryGroupThrottlingKey;
+            }
+
+            @Override
+            public void onFailure(String source, Exception e) {
+                logger.warn("Failed to delete QueryGroup due to error: {}, for source: {}", e.getMessage(), source);
+                listener.onFailure(e);
+            }
+
+            @Override
+            public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                final Map<String, QueryGroup> oldGroupsMap = oldState.metadata().queryGroups();
+                final Map<String, QueryGroup> newGroupsMap = newState.metadata().queryGroups();
+
+                List<QueryGroup> deletedGroups = oldGroupsMap.keySet()
+                    .stream()
+                    .filter(groupId -> !newGroupsMap.containsKey(groupId))
+                    .map(oldGroupsMap::get)
+                    .collect(Collectors.toList());
+
+                DeleteQueryGroupResponse response = new DeleteQueryGroupResponse(deletedGroups, RestStatus.OK);
+                listener.onResponse(response);
```

Comment:
This will even collect the query groups not deleted by this request and return as response to the client. We should only return the query group deleted as part of this request.